routes/wishlists.router.js

```diff
@@ -0,0 +1,77 @@
+const express = require("express");
+const router = express.Router();
+const { Wishlist } = require("../models/wishlist.model");
+const { User } = require("../models/user.model");
+
+
+router.param("userid", async(req, res, next, id)=>{
+    try{
+        const user = await User.findById({_id: id});
+        
+        if(!user){
+            res.status(404).json({success:false, message: "No user found associated, please check the user id!"});
+            return;
+        }
+        req.user = user;
+        next();
+    } catch(error) {
+        res.status(500).json({success:false, message: "Request failed please check errorMessage key for more details", errorMessage: error.message })
+    }
+})
+router.param("userid", async(req,res,next,id)=>{
+    try {
+        let wishlist = await Wishlist.findOne({userId: id});
+        
+        if(!wishlist){
+            wishlist = new Wishlist({userId:id, products: []});
+            wishlist = await wishlist.save();
+        }
+        req.wishlist = wishlist;
+        next();
+    } catch(error) {
+        res.status(500).json({success:false, message: "Request failed please check errorMessage key for more details", errorMessage: error.message })
+    }
+})
+
+router.route("/:userid/wishlist")
+.get( async(req,res)=>{
+    try { 
+        let {wishlist} = req ;
+        wishlist = await wishlist.populate({path:"products.productId", select: 'name price image brand offer inStock url',}).execPopulate();
+        res.status(200).json({ response : wishlist, success : true });
+
+    }  catch(error) {
+        console.error(error)
+        res.status(500).json({success:false, message: "Request failed please check errorMessage key for more details", errorMessage: error.message })
+    }
+})
+.post( async(req,res)=>{
+    try {       
+        const productUpdates = req.body;
+        const {wishlist} = req ;
+        
+        const isProductAlreadyAdded = wishlist.products.find((product)=>product.productId == productUpdates._id);
+        
+        // if product is already present in the wishlist then toggling the status of product
+        if(isProductAlreadyAdded) {
+           for( let product of wishlist.products) {
```

Comment:
could we filter and map instead , I am still thinking about the same

Reply:
Yes, it would be more readable, updated it